docs/source/pytest.rst

```diff
@@ -0,0 +1,358 @@
+       """
+       hdl.sources = (
+           # List HDL source files,
+           "sample_module.sv",
+       )
+
+       return hdl
+
+
+Example content of ``test_sample_module.py`` file:
+
+.. code:: python
+
+   import pytest
+   from cocotb_tools.pytest.hdl import HDL
+
+
+   # Without providing positional arguments or test_module option to cocotb decorator,
+   # plugin will use current file as cocotb testbench (Python file with cocotb tests).
+   # If toplevel option was not provided, it will be based on name of first test_module
+   # but with removed test_* prefix or *_test suffix.
+   @pytest.mark.cocotb  # equivalent to @pytest.mark.cocotb("test_dut", toplevel="dut")
+   def test_sample_module(sample_module: HDL) -> None:
+       """Build HDL design and run HDL simulator to execute cocotb tests.
+
+       Args:
+           sample_module: Instance of defined HDL design.
+       """
+       sample_module.test()
+
+
+   # @pytest.mark.cocotb or @cocotb.test decorator is not required if test function
+   # starts with test_* prefix, is coroutine function (``async``) and with ``dut`` argument.
+   async def test_some_dut_feature(dut) -> None:
+        """Cocotb test for DUT."""
+
+@pytest.mark.cocotb
+===================
+
+Provided ``@pytest.mark.cocotb`` marker by :py:mod:`cocotb_tools.pytest.plugin` allows
+to configure all aspects of cocotb test and cocotb runner. Marker recognizes all
+named arguments from :py:func:`cocotb.test` and :py:class:`cocotb_tools.runner.Runner`.
+Additionally, positional arguments of ``@pytest.mark.cocotb`` marker are equivalent to
+``test_module`` argument from :py:func:`cocotb.test`.
+
+If no positional arguments were provided to ``@pytest.mark.cocotb`` or ``test_module`` argument is empty/non-set,
+plugin will load current Python module where ``@pytest.mark.cocotb`` was used as cocotb testbench (Python file with
+cocotb tests).
+
+If ``toplevel`` argument is empty/non-set, plugin will use name of first test module but without
+``test_*`` prefix or ``*_test`` suffix. For example, if test module was ``test_dut`` then
+name of HDL top level design will be ``dut``.
+
+Using ``@pytest.mark.cocotb`` marker to mark test function as cocotb test is optional
+for test functions that are starting with ``test_*`` prefix name, are coroutine functions (``async def``) and
+with ``dut`` argument. Normal functions (non-coroutines) with ``@pytest.mark.cocotb`` marker are
+marked as cocotb runner that should run HDL simulator by invoking
+:py:func:`cocotb_tools.pytest.hdl.HDL.test`, :py:func:`cocotb_tools.runner.Runner.test` or similar method.
+
+Marker can also help plugin to identify and bind cocotb tests to cocotb runners. This is done by plugin
+based on information from provided positional arguments (or cocotb ``test_module`` argument) supplied into
+``@pytest.mark.cocotb`` decorator. This helps plugin to properly filter tests out
+when using `pytest`_ ``-k '<expression>'`` or ``-m '<markers>'`` options.
```

Comment:
There's a lot of explaining before we get into an example, is it possible to add another example here?

Reply:
(I skipped reviewing this part for now)

Reply:
@ktbarrett Added more examples in this commit 40241d0e7961e9ca4c1b33d95886a381dcb37011